test/scripts/SapphireIntegration.test.ts

```diff
@@ -0,0 +1,260 @@
+import * as sapphire from '@oasisprotocol/sapphire-paratime'
+import addresses from '@oceanprotocol/contracts/addresses/address.json'
+import { ethers } from 'ethers'
+import { AccesslistFactory } from '../../src/contracts/AccessListFactory'
+import { AccessListContract } from '../../src/contracts/AccessList'
+import { NftFactory } from '../../src/contracts/NFTFactory'
+import { ZERO_ADDRESS } from '../../src/utils/Constants'
+import { assert } from 'console'
+import { Datatoken4 } from '../../src/contracts/Datatoken4'
+import { AbiItem, Config, Nft, NftCreateData } from '../../src'
+import ERC20Template4 from '@oceanprotocol/contracts/artifacts/contracts/templates/ERC20Template4.sol/ERC20Template4.json'
+import { getEventFromTx } from '../../src/utils'
+
+describe('Sapphire tests', async () => {
+  const provider = sapphire.wrap(
+    ethers.getDefaultProvider(sapphire.NETWORKS.testnet.defaultGateway)
+  )
+  const wallet = new ethers.Wallet(process.env.PRIVATE_KEY, provider)
```

Comment:
Given that these tests are using a private key on a live network, at some point, the wallet address is going to run out of tokens and the tests will fail. Is there some sort fo longer-term plan here, like getting the tests to work with barge? If so, do we have an issue for this?

Reply:
That is the reason that I did not include the tests in the CI, because everytime the account will be drained, we'll have to top up the funds for those. For the moment it's not tested with CI, only locally. We can create an issue for it. @mihaisc suggested to have a separate job which not runs at every push but he said to leave this problem for now, I'll create an issue for it.

Reply:
ok cool, fair enough